smsapi/Api/Action/Contacts/ContactAdd.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SMSApi\Api\Action\Contacts;
+
+use DateTime;
+use InvalidArgumentException;
+use SMSApi\Api\Response\Contacts\ContactResponse;
+use SMSApi\Client;
+use SMSApi\Proxy\Proxy;
+
+final class ContactAdd extends ContactsAction
+{
+    private static $PARAMETER_METHODS = [
+        ContactResponse::FIELD_FIRST_NAME => 'setFirstName',
+        ContactResponse::FIELD_LAST_NAME => 'setLastName',
+        ContactResponse::FIELD_PHONE_NUMBER => 'setPhoneNumber',
+        ContactResponse::FIELD_EMAIL => 'setEmail',
+    ];
+
+    public static function fromFirstName($firstName, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_FIRST_NAME, $firstName, $client, $proxy);
+    }
+
+    public static function fromLastName($lastName, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_LAST_NAME, $lastName, $client, $proxy);
+    }
+
+    public static function fromPhoneNumber($phoneNumber, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_PHONE_NUMBER, $phoneNumber, $client, $proxy);
+    }
+
+    public static function fromEmail($email, Client $client, Proxy $proxy)
+    {
+        return new self(ContactResponse::FIELD_EMAIL, $email, $client, $proxy);
+    }
+
+    /**
+     * @param string $parameterName
+     * @param string|int $parameterValue
+     * @param Client $client
+     * @param Proxy $proxy
+     */
+    public function __construct($parameterName, $parameterValue, Client $client, Proxy $proxy)
+    {
+        parent::__construct($client, $proxy);
+
+        $this->{self::$PARAMETER_METHODS[$parameterName]} = $parameterValue;
+    }
+
+    public function getMethod()
+    {
+        return self::METHOD_POST;
+    }
+
+    protected function getResource()
+    {
+        return '/contacts';
+    }
+
+    protected function response($data)
+    {
+        return ContactResponse::fromJsonString($data);
+    }
+
+    public function setPhoneNumber($phoneNumber)
```

Comment:
If I use `ContactAdd::fromPhoneNumber` to create object then I assume that your intention was to have it in proper state.
Exposing `setPhoneNumber` and other setters may cause without any validation may cause object to fall into invalid state.